MachineLearning/ibmdbpy_AssociationRulesMining.ipynb

```diff
@@ -0,0 +1,1807 @@
+    "\n",
+    "__Contents__\n",
+    "\n",
+    "This notebook uses the transactions dataset of an outdoor store. Let's see what patterns we find!\n",
+    "\n",
+    "__1. Get started__\n",
+    "* Import what you need\n",
+    "* Establish a JDBC connection to Db2 Warehouse\n",
+    "* Explore the data\n",
+    "    \n",
+    "__2. Mine Association Rules with ibmdbpy__\n",
+    "* Reminder about Association Rules Mining\n",
+    "    * General task\n",
+    "    * Definitions\n",
+    "* Put your knowledge into practice\n",
+    "    * Define an AssociationRules object\n",
+    "    * Analyze the results\n",
+    "    * Experiment for yourself with pruning!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Get started"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "__Import what you need__\n",
+    "\n",
+    "As usual, we first import some basic modules from ibmdbpy."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ibmdbpy import IdaDataBase, IdaDataFrame"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "__Establish a JDBC connection to Db2 Warehouse__\n",
+    "\n",
+    "Please enter your credentials in order to connect to the database."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Enter the values for you database connection\n",
+    "dsn_database = \"___\" # e.g. \"BLUDB\"\n",
+    "dsn_hostname = \"___\" # e.g.: \"abc.url.example\"\n",
+    "dsn_port = \"___\"    # e.g. \"50000\"\n",
+    "dsn_uid = \"___\"     # e.g. \"db2_1234\"\n",
+    "dsn_pwd = \"___\"     # e.g. \"zorglub\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Establish connection\n",
+    "connection_string='jdbc:db2://'+dsn_hostname+':'+dsn_port+'/'+dsn_database+':user='+dsn_uid+';password='+dsn_pwd+\";\" \n",
+    "idadb=IdaDataBase(dsn=connection_string, verbose = False)\n",
+    "# Set verbose to True if you want to see the detail of ibmdbpy operations"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "By default the environment variable `AUTOCOMMIT` is then set to True, which means that every SQL statement which is submitted through the connection is executed within its own transaction and then committed implicitly. When you close the connection to Db2, if the environment variable `AUTOCOMMIT` is set to False, then all changes after the last explicit commit are discarded. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "__Explore the data__\n",
+    "\n",
+    "We use sample data available out of the box in Db2 Warehouse. \n",
```

Comment:
The sample data is not available out of the box. It needs to be installed explicitly, see https://www.ibm.com/support/knowledgecenter/SSCJDQ/com.ibm.swg.im.dashdb.doc/admin/sampledata.html (you can just add a link to this in the `Prerequisites` section. It's only available on the machine you've been using because the install the sample container automatically.

Reply:
yes right, I have added the link to the prerequisites